lib/portlayer/exec/container_cache.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"sync"
+)
+
+/*
+* ContainerCache will provide an in-memory cache of containerVMs.  It will
+* be refreshed on portlayer start and updated via container lifecycle
+* operations (start, stop, rm) and well as in response to infrastructure
+* events
+ */
+type containerCache struct {
+	m sync.RWMutex
+
+	// cache maps containerVM to container ID && infrastructure ID
+	cacheByID      map[string]*Container
```

Comment:
We are storing id -- which is randomly generated and follows the the regex of [a-z0-9] -- originally I was checking via regex to classify as either id or ref and store in the appropriate map.  This is not the friendly name that a user can specify...

The moref will be the full moref so Object:ID -- "VirtualMachine:vm-100" -- again I think that a collision is doubtful.